lang/python/python/patches/024-bpo-35907-Complete-test_urllib.test_local_file_open-GH-13506.patch

```diff
@@ -0,0 +1,43 @@
+From c70e9880446e9e9149e4a314821d380673d0332c Mon Sep 17 00:00:00 2001
+From: Victor Stinner <vstinner@redhat.com>
+Date: Wed, 22 May 2019 22:30:10 +0200
+Subject: [PATCH] bpo-35907: Complete test_urllib.test_local_file_open()
+
+Test also URLopener().open(), URLopener().retrieve(), and
+DummyURLopener().retrieve().
+---
+ Lib/test/test_urllib.py                                     | 6 +++++-
+ .../next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst   | 4 +++-
+ 2 files changed, 8 insertions(+), 2 deletions(-)
+
+diff --git a/Lib/test/test_urllib.py b/Lib/test/test_urllib.py
+index ae1f6c0b29f0..22b0874a9281 100644
+--- a/Lib/test/test_urllib.py
++++ b/Lib/test/test_urllib.py
+@@ -1049,12 +1049,16 @@ def open_spam(self, url):
+             "//c:|windows%/:=&?~#+!$,;'@()*[]|/path/")
+ 
+     def test_local_file_open(self):
++        # bpo-35907, CVE-2019-9948: urllib must reject local_file:// scheme
+         class DummyURLopener(urllib.URLopener):
+             def open_local_file(self, url):
+                 return url
+         for url in ('local_file://example', 'local-file://example'):
+-            self.assertRaises(IOError, DummyURLopener().open, url)
+             self.assertRaises(IOError, urllib.urlopen, url)
++            self.assertRaises(IOError, urllib.URLopener().open, url)
++            self.assertRaises(IOError, urllib.URLopener().retrieve, url)
++            self.assertRaises(IOError, DummyURLopener().open, url)
++            self.assertRaises(IOError, DummyURLopener().retrieve, url)
+ 
+ # Just commented them out.
+ # Can't really tell why keep failing in windows and sparc.
+diff --git a/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst b/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst
+index bb187d8d65a5..6a448ce6261c 100644
+--- a/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst
++++ b/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst
+@@ -1 +1,3 @@
+-CVE-2019-9948: Avoid file reading as disallowing the unnecessary URL scheme in urllib.urlopen
++CVE-2019-9948: Avoid file reading as disallowing the unnecessary URL scheme in
```

Comment:
i would not add the patches don't change lib functionality;
they are just noise
a lot [or all] of the tests are not even packaged;

Reply:
i am mostly referring to patch `[PATCH] bpo-35907: Complete test_urllib.test_local_file_open()`

Reply:
Actually I'd like to see the tests packaged at some point (for both the Python interpreter and for libraries), but this is just a far away goal. I'm ok with removing the tests patch.

You are ok with the patches that only affect NEWS? :joy:

Reply:
> You are ok with the patches that only affect NEWS? joy

I'd remove them if they only affect NEWS.
We will get them eventually when we upgrade.
Thing is, I would not add stuff (as patches) if we get them as upgrade.

Reply:
Ok :smile: